src/core/ucc_team.c

```diff
@@ -121,8 +125,14 @@ ucc_status_t ucc_team_create_post(ucc_context_h *contexts, uint32_t num_contexts
         status = UCC_ERR_NO_MEMORY;
         goto err_ctx_alloc;
     }
+
     memcpy(team->contexts, contexts, sizeof(ucc_context_t *) * num_contexts);
     ucc_copy_team_params(&team->params, params);
+    /* check if user provides team id and if it is not too large */
+    if ((params->mask & UCC_TEAM_PARAM_FIELD_ID) &&
+        (params->id <= UCC_TEAM_ID_MAX)) {
```

Comment:
why is param->id is 64 bit and internal ID is 16 bit? what happens if user creates two teams with MSB bit sets (LSB 15 bits 0)?

Reply:
Currently we have a limitation on internal team id: 16 bits, due to UCP tag space limitation. In the API we decided to keep it uint64_t since maybe we will be able to support larger IDs in the future. Here we check if the value provided by the user can be actually used. If MSB bits are set then params->id > UCC_TEAM_ID_MAX. We will not use it: we will allocate team id internally. Only when params->id fits into 15 bits, we will take it (e.g., in hpcx in ompi this is the case for many many workloads).